src/app/WriteHandler.cpp

```diff
@@ -138,11 +138,17 @@ CHIP_ERROR WriteHandler::OnMessageReceived(Messaging::ExchangeContext * apExchan
                        "Incoming exchange context should be same as the initial request.");
     VerifyOrDieWithMsg(!apExchangeContext->IsGroupExchangeContext(), DataManagement,
                        "OnMessageReceived should not be called on GroupExchangeContext");
+
     if (!aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::WriteRequest))
     {
         ChipLogDetail(DataManagement, "Unexpected message type %d", aPayloadHeader.GetMessageType());
+        if (!aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::StatusResponse))
+        {
+            err = StatusResponse::Send(Status::InvalidAction, apExchangeContext,
+                                       false /*aExpectResponse*/);
```

Comment:
Again: if this succeeds, mpExchangeCtx is now dangling. Need to null it out on success.

This was an issue in the previous version too....

Reply:
Done, thanks